### DISCUSSION TITLE
Point subdomain at Nextjs page via Vercel DNS.

### DISCUSSION THREAD

**Discussion by lmf-git** [vercel/next.js#41017] [Help]
Point subdomain at Nextjs page via Vercel DNS:

```
async rewrites() {
	return [
		{
			source: 'https://:postOrProgramId*.tikex-dev.com/',
			destination: 'https://tikex-dev.com/:postOrProgramId*',
		},
	]
},
```

I would do rewrite like this `amazing-campaign-in-nyc.tikex-dev.com` -> `tikex-dev.com/amazing-campaign-in-nyc`

Unfortunatelly when I run locally, I got this error, though I know it does not support localhost now.

```
`source` does not start with / for route {"source":"https://:postOrProgramId*.tikex-dev.com/","destination":"https://tikex-dev.com/:postOrProgramId*"}
```

I am not sure, I need `:` before the variable and `*` after it? How can I express subdomain to subpath redirect?

I have an other reverseproxy rewrite. Do I have to worry about the two will interfere?

```
{
	source: '/api/:path*',
	destination: `${
		true
			? `https://api.tikex-${
					process.env.NEXT_PUBLIC_ENVIRONMENT ?? 'dev'
			  }.com/`
			: 'http://localhost:8080/'
	}:path*`,
},
```

_Originally posted by @j4nos in https://github.com/vercel/next.js/discussions/38733_

**Comment by lmf-git** [comment]
Bad documentation for reasonable use case of: Point subdomain at Nextjs page via Vercel DNS.

**Comment by ijjk** [comment]
Hi, it sounds like you are trying to do a `has` rewrite which has related documentation here https://nextjs.org/docs/api-reference/next.config.js/rewrites#header-cookie-and-query-matching

**Comment by lmf-git** [comment]
I'd already read it and every stackoverflow question on the topic before
posting the issue. It isn't useful to me.

I have an app which has a page called /magic

I want to make a subdomain (magic.example.com) point at this page.

Please tell me what configuration is required on Vercel and Nextjs to do
this, it seems like a straightforward thing that should be supported if you
have these features?

On Tue, Sep 27, 2022 at 8:50 PM JJ Kasper ***@***.***> wrote:

> Hi, it sounds like you are trying to do a has rewrite which has related
> documentation here
> https://nextjs.org/docs/api-reference/next.config.js/rewrites#header-cookie-and-query-matching
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vercel/next.js/issues/40964#issuecomment-1259976483>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBYVODTCX3DEHPZSP5PH7DWANFZNANCNFSM6AAAAAAQXCDRKQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
LM Fielding

Founder

Dreamsbyte
+44 738-980-5421 <+44+738-980-5421>
***@***.***
https://dreamsbyte.com
[image: facebook] <https://www.facebook.com/dreamsbyte>
[image: linkedin] <https://linkedin.com/company/dreamsbyte>
[image: instagram] <https://www.instagram.com/dreamsbyte/>

**Comment by ijjk** [comment]
I think the route you are looking for is:

```js
{
  source: '/',
  destination: '/:postOrProgramId*',
  has: [
    {
      type: 'host',
      value: '(?<postOrProgramId>.*).tikex-dev.com'
    }
  ]
}
```

Note: if you have a `pages/index.js` file you will need to return the above rewrite in `beforeFiles` as documented. 

If you have any suggestions on how we can make this more clear in the docs would be happy to hear them!
